modules/scratchFetch.ts

```diff
@@ -0,0 +1,242 @@
+							},
+						],
+						thumbnail: {
+							url: projectdata.images["282x218"],
+						},
+						author: {
+							name: projectdata.author.username,
+							url: `https://scratch.mit.edu/users/${projectdata.author.username}`,
+							icon_url: projectdata.author.profile.images["90x90"],
+						},
+						footer: {
+							text: notSet ? "Disable this using /settings" : "",
+						},
+						url: `https://scratch.mit.edu/projects/${urlParts[4]}`,
+					},
+				],
+				allowedMentions: { repliedUser: false },
+			});
+
+			break;
+		case "users":
+			const userdata = await fetchApiData(
+				`https://${constants.urls.scratchApi}/users/${urlParts[4]}/`,
+			).catch(() => {});
+
+			message.reply({
+				embeds: [
+					{
+						title: userdata.username,
+						description: "",
+						color: 0x885cd4,
+
+						fields: [
+							{
+								name: `About`,
+								value: userdata.profile.bio,
+							},
+							{
+								name: `WIWO`,
+								value: userdata.profile.status,
+							},
+						],
+						thumbnail: {
+							url: userdata.profile.images["90x90"],
+						},
+						author: {
+							name: userdata.scratchteam ? "Scratch Team" : "",
+						},
+						footer: {
+							text: notSet ? "Disable this using /settings" : "",
+						},
+						url: `https://scratch.mit.edu/users/${urlParts[4]}`,
+					},
+				],
+				allowedMentions: { repliedUser: false },
+			});
+
+			break;
+		case "studios":
+			const studiodata = await fetchApiData(
+				`https://${constants.urls.scratchApi}/studios/${urlParts[4]}/`,
+			).catch(() => {});
+			message.reply({
+				embeds: [
+					{
+						title: studiodata.title,
+						description: long(studiodata.description, 400, "..."),
+						color: 0x885cd4,
+
+						fields: [
+							{
+								name: `comments`,
+								value: studiodata.comments_allowed
+									? `${studiodata.stats.comments}`
+									: `${studiodata.stats.comments} (off)`,
+								inline: true,
+							},
+							{
+								name: `followers`,
+								value: studiodata.stats.followers,
+								inline: true,
+							},
+							{
+								name: `managers`,
+								value: studiodata.stats.managers,
+								inline: true,
+							},
+							{
+								name: `projects`,
+								value: studiodata.stats.projects,
+								inline: true,
+							},
+						],
+						thumbnail: {
+							url: studiodata.image,
+						},
+
+						footer: {
+							text: notSet ? "Disable this using /settings" : "",
+						},
+						url: `https://scratch.mit.edu/studios/${urlParts[4]}`,
+					},
+				],
+				allowedMentions: { repliedUser: false },
+			});
+
+			break;
+		case "discuss":
+			switch (urlParts[4]) {
+				case "topic":
+					const topicdata = await fetchApiData(
+						`https://scratchdb.lefty.one/v3/forum/topic/info/${urlParts[5]}/`,
+					).catch(() => {});
+					const topicposts = await fetchApiData(
+						`https://scratchdb.lefty.one/v3/forum/topic/posts/${urlParts[5]}/`,
+					).catch(() => {});
+					const topicfirst = topicposts[0];
```

Comment:
Actually, we only need one fetch, because `topicfirst.topic` has all the same info as `topicdata`. The important stuff at least. The only thing we'd have to get rid of is total post count, which is fine.
Could `topicposts` be renamed to `posts` and `topicfirst` to `post`?
Also, the posts fetch needs `?o=oldest` at the end